src/pages/search.js

```diff
@@ -0,0 +1,146 @@
+      <h2 className="hed-primary">Search{hitsInfo}</h2>
+      <form>
+        <label htmlFor="item">
+          Search
+          <input
+            type="text"
+            aria-label="Search"
+            id="item"
+            onChange={event => {
+              clearTimeout(searchEvent)
+              const { value } = event.currentTarget
+              searchEvent = setTimeout(() => {
+                setQuery(value)
+              }, 500)
+            }}
+          />
+        </label>
+        <button
+          type="submit"
+          onClick={async event => {
+            event.preventDefault()
+            await querySearch(searchState, searchDispatch)
+          }}
+        >
+          Search
+        </button>
+      </form>
+      {searchState.isFetching && (
+        <div>
+          <img
+            src="https://d1j8pt39hxlh3d.cloudfront.net/products/previews/RES3POBSZ353HFVPZOKR/2329_kKbUTLGEVXhuOIJqrT7QvIJFAXvEpQ3z.gif"
+            alt="Searching..."
+          />
+          <h3>Searching...</h3>
+        </div>
+      )}
+      <div className={searchPageStyle.searchResults}>
+        {/* State results */}
+        {results.states.nbHits > 0 && !searchState.isFetching && (
+          <div className={searchPageStyle.searchResultsSection}>
+            <h3>States ({results.states.nbHits})</h3>
+            {results.states.hits.map(state => (
+              <div key={state.state} className={searchPageStyle.searchResult}>
+                <PublicationTitle>
+                  <Link to={`${state.slug}`}>{state.name}</Link>
+                </PublicationTitle>
+              </div>
+            ))}
+          </div>
+        )}
+
+        {/* Blog post results */}
+        {results.blogPosts.nbHits > 0 && !searchState.isFetching && (
+          <div className={searchPageStyle.searchResultsSection}>
+            <h3>Blog Posts ({results.blogPosts.nbHits})</h3>
+            {results.blogPosts.hits.map(post => (
+              <div key={post.objectID} className={searchPageStyle.searchResult}>
+                <PublicationTitle>
+                  <Link to={`/blog/${post.slug}`}>{post.title}</Link>
+                </PublicationTitle>
+                <DetailText>
+                  {post.author_name}
+                  <span className={pressListStyle.dotSeparator}>•</span>
+                  {post.publishDate}
+                </DetailText>
+              </div>
+            ))}
+          </div>
+        )}
+
+        {/* Pages results */}
+        {results.pages.nbHits > 0 && !searchState.isFetching && (
+          <div className={searchPageStyle.searchResultsSection}>
+            <h3>Pages ({results.pages.nbHits})</h3>
+            {results.pages.hits.map(page => (
+              <div key={page.objectID} className={searchPageStyle.searchResult}>
+                <PublicationTitle>
+                  {/* FIXME this should be handled during indexing 
+                  (avoids external Link issues) */}
```

Comment:
This is tricky...it looks like _some_ `Link`s begin with '/', and others don't. We should probably standardize this across the codebase so we don't have to handle this here. Ideally someone could work some `grep` magic and find out where that's happening.

@kevee which is preferred?
1. prefixing `Link` with `/`
2. *not* prefixing `Link` with `/`

Reply:
Actually we *have* to prefix links with `/` otherwise Gatsby will throw warnings.
I'm pretty confident this is an inconsistency issue with the slugs and how these are built.

I could comment out this function to get a precise list of failing links, so we can check these individually

Reply:
Prefix them all with `/`

The pages here all have their slug defined in Contentful, which automatically creates the slug based on the title.

Reply:
Added `getSanitizedSlug` to generate slug depending on result type + handle missing starting `/`